CONTRIBUTING.md

```diff
@@ -1,128 +1,128 @@
+    |   |-- production.js
+    |   |-- staging.js
+    |   |-- test.js
+    |-- controllers
+    |   |-- healthController.js
+    |   |-- // Controller files concerning function on a similar entity
+    |-- logs
+    |   |-- // log files
+    |-- middlewares
+    |   |-- // individual middleware files to be required on server start or in the route middleware
+    |-- models
+    |   |-- // Files consisting of the individual table/collection config and wrapper interaction functions
+    |-- routes
+    |   |-- index.js // routes files separated by their first path string
+    |   |-- auth.js // the individual routes files should contain the OPEN API JSDOC reference
+    |-- services
+    |   |-- authService.js // Files using any 3rd party library/service or providing any secific service in the project
+    |-- scripts // Standalone scripts
+    |-- test
+    |   |-- fixtures
+    |   |   |-- auth
+    |   |       |-- githubUserInfo.js
+    |   |-- integration // Integration tests
+    |   |   |-- authController.test.js
+    |   |-- unit // Unit tests
+    |   |   |-- middlewares
+    |   |   |-- services
+    |   |-- utils // Utility functions to be used while testing
+    |-- utils // Files containing utility functions
+    |    |-- logger.js
+    |-- .github
+    |   |-- workflows
+    |       |-- // Github actions files
+    |-- .gitignore
+    |-- .*rc, .*js, .*json, .*yml // config files for dependencies
+    |-- CONTRIBUTING.md
+    |-- README.md
+    |-- CHANGELOG.md
+    |-- app.js
+    |-- firestore-private-key.json // Firestore key file
+    |-- package-lock.json
+    |-- package.json
+    |-- server.js // Contains server start logic
+
+```
+
+## Generating Authentication Token
+- Run the project locally, make sure the server is listening to requests
+- Navigate to `https://github.com/login/oauth/authorize?client_id=<GITHUB_CLIENT_ID>`
+- Authorize the application
+- Copy the response cookie named rds-session from the redirected request <HOST>/auth/github/callback?code=<SOME_AUTOGENERATED_CODE>
+- Use the cookie for authenticated routes in the API.
+- For non-production environments, authentication is also supported with the `Authorization` header.
+- Authorization header: `Authorization: Bearer <token>`
+
+## Testing Guidelines
+- Libraries used in testing in the project:
+  - [mocha](https://mochajs.org/): Test framework
+  - [chai](https://www.chaijs.com/): Assertion library
+  - [sinon](https://sinonjs.org/): JavaScript test spies, stubs and mocks
+  - [nock](https://github.com/nock/nock/blob/main/README.md): HTTP requests mocking
+- The test suite uses [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite) for running firestore for tests([documentation](https://firebase.google.com/docs/emulator-suite/install_and_configure)).
+- Pre-requisites:
+  - Node.js version 8.0 or higher.
+  - Java version 1.8 or higher.
+  - Firebase CLI, mocha and nyc: `npm install -g firebase-tools mocha nyc`

```

Comment:
Can we please explicitly mention that for Windows, Git Bash is recommended to run the test scripts?
Also, the `test` script won't work for windows. Windows users will have to run `test-integration` first and then `test-unit` manually. This is because when the `test` script is run, the emulator window doesn't close by itself after running integration test and gives error that PORT 8080 is blocked by some process (which is the emulator window used for integration tests) and unit tests won't start.

Reply:
Yeah good idea! It will be helpful. I will do the changes accordingly. Thanks for the review! 😄